pyadjoint/ordered_set.py

```diff
@@ -0,0 +1,134 @@
+from typing import Hashable, Iterable
+
+
+class OrderedSet(set):
```

Comment:
Seems more natural to inherit from `collections.abc.MutableSet`.

Reply:
Ah yeah, I always forget about those. Especially with dictionary backing, this seems more reasonable.